test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -849,7 +849,7 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 		framework.ExpectNoError(e2enode.WaitForReadyNodes(c, nodeCount-minSize+1, resizeTimeout))
 		ngNodes, err := framework.GetGroupNodes(minMig)
 		framework.ExpectNoError(err)
-		gomega.Expect(len(ngNodes) == 1).To(gomega.BeTrue())
+		framework.ExpectEqual(len(ngNodes), 1)
```

Comment:
I am good at this change, because here makes the code easy.
But TBH I don't have strong opinion about the other parts in this PR.